openrouteservice/src/test/java/org/heigit/ors/routing/traffic/TrafficSpeedCalculatorTest.java

```diff
@@ -0,0 +1,121 @@
+        IntsRef edgeFlags = encodingManager.createEdgeFlags();
+        double originalEdgeSpeed = 40.0;
+        int edgeId = 2;
+        carEncoder.getAverageSpeedEnc().setDecimal(false, edgeFlags, originalEdgeSpeed);
+        EdgeIteratorState edgeIteratorState = createMockedEdgeIteratorState(10, edgeFlags, 0, 1, edgeId, 2, 3);
+        assertEquals(38, trafficSpeedCalculator.getSpeed(edgeIteratorState, false, 1), 1e-8);
+    }
+
+    @Test
+    public void testOriginalLargerThan45AndTrafficSlower() {
+        IntsRef edgeFlags = encodingManager.createEdgeFlags();
+        double originalEdgeSpeed = 60.0;
+        int edgeId = 3;
+        carEncoder.getAverageSpeedEnc().setDecimal(false, edgeFlags, originalEdgeSpeed);
+        EdgeIteratorState edgeIteratorState = createMockedEdgeIteratorState(10, edgeFlags, 0, 1, edgeId, 2, 3);
+        assertEquals(50, trafficSpeedCalculator.getSpeed(edgeIteratorState, false, 1), 1e-8);
+    }
+
+    @Test
+    //Do not overwrite speeds slower than 45 in case traffic is faster
+    public void testOriginalSmallerThan45AndTrafficFaster() {
+        IntsRef edgeFlags = encodingManager.createEdgeFlags();
+        double originalEdgeSpeed = 40.0;
+        int edgeId = 4;
+        carEncoder.getAverageSpeedEnc().setDecimal(false, edgeFlags, originalEdgeSpeed);
+        EdgeIteratorState edgeIteratorState = createMockedEdgeIteratorState(10, edgeFlags, 0, 1, edgeId, 2, 3);
+        assertEquals(40, trafficSpeedCalculator.getSpeed(edgeIteratorState, false, 1), 1e-8);
+    }
+
+    @Test
+    //Do not overwrite speed data if traffic data is much faster than 110% of original speed
+    public void testOriginalLargerThan45AndTrafficMuchFaster() {
+        IntsRef edgeFlags = encodingManager.createEdgeFlags();
+        double originalEdgeSpeed = 60.0;
+        int edgeId = 5;
+        carEncoder.getAverageSpeedEnc().setDecimal(false, edgeFlags, originalEdgeSpeed);
+        EdgeIteratorState edgeIteratorState = createMockedEdgeIteratorState(10, edgeFlags, 0, 1, edgeId, 2, 3);
+        assertEquals(60, trafficSpeedCalculator.getSpeed(edgeIteratorState, false, 1), 1e-8);
+    }
+
+    @Test
+    //Increase speed only within 110% of original speed
+    public void testOriginalLargerThan45AndTrafficFaster() {
+        IntsRef edgeFlags = encodingManager.createEdgeFlags();
+        double originalEdgeSpeed = 60.0;
+        int edgeId = 6;
+        carEncoder.getAverageSpeedEnc().setDecimal(false, edgeFlags, originalEdgeSpeed);
+        EdgeIteratorState edgeIteratorState = createMockedEdgeIteratorState(10, edgeFlags, 0, 1, edgeId, 2, 3);
+        assertEquals(65, trafficSpeedCalculator.getSpeed(edgeIteratorState, false, 1), 1e-8);
+    }
+
+    private class MockTrafficStorage extends TrafficGraphStorage {
+
+        @Override
+        public int getSpeedValue(int edgeId, int baseNode, int adjNode, long unixMilliSeconds, int timeZoneOffset) {
+            if (edgeId == 1)
```

Comment:
Replacing this if-chain with switch would make the code more readable and more performant.

Reply:
Don't think this changed anything about readability or performance, but I replaced it.

Reply:
Thanks. Concerning readability: With a switch you immediately know that the variable you are branching on is always the same and you communicate that this is the intended bahaviour. In an if-else-chain you have to check each if-statement to know this fact. Concering performance: the compiler creates a lookup-table to know where to jump to when switch is used. In the below example if edgeId=6 the processor must perform all previous if-instructions.